panc/src/main/java/org/quattor/pan/dml/data/ProtectedListResource.java

```diff
@@ -25,7 +25,11 @@ public Element duplicate() {
 
 	@Override
 	public Element get(Term key) throws InvalidTermException {
-		return baseList.get(key);
+		final Element value = baseList.get(key);
```

Comment:
It does not make a copy - both ProtectedHashResource and ProtectedListResource are just wrappers which say "you can't modify the data through this reference", but the data is not copied.

The copying happens when the resource is used - there are lots of places with logic like "check if the resource is protected, and if it is, make a copy and work with that instead".